xbmc/cores/RetroPlayer/streams/RetroPlayerAudio.cpp

```diff
@@ -118,8 +122,19 @@ void CRetroPlayerAudio::AddStreamData(const StreamPacket &packet)
   {
     if (m_pAudioStream)
     {
+      double delay = m_pAudioStream->GetDelay();
+
       const size_t frameSize = m_pAudioStream->GetChannelCount() * (CAEUtil::DataFormatToBits(m_pAudioStream->GetDataFormat()) >> 3);
-      m_pAudioStream->AddData(&audioPacket.data, 0, static_cast<unsigned int>(audioPacket.size / frameSize));
+
+      unsigned int frameCount = static_cast<unsigned int>(audioPacket.size / frameSize);
+
+      if (delay > 0.3)
+      {
+        m_pAudioStream->Flush();
+        CLog::Log(LOGDEBUG, "RetroPlayer[AUDIO]: Audio delay (%0.4f seconds) is too high - flushing", delay);
```

Comment:
Would it make sense to log in units of milliseconds? that's what the game loop currently does